### PR TITLE
Randomize server name in test_sql_server_mgmt.

### PR DESCRIPTION
**Pull request by jaredmoo**
It seems a littly hacky to use SqlServerPreparer just to generate server name, but I couldn't see how to use SingleValueReplacer on its own.

I also ran into issues where the two SqlServerPreparers were generating the same random name - this seemed to be because they were running so quickly (due to not actually creating a resource) that utilities.create_random_name was getting the same timestamp twice in a row. So I added RANDOM_NAME_COUNT into the mix in order to prevent duplicate random names.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by troydai** [comment]
It is not ideal to introduce a global variable for the sake of a unique name. There are other approaches. For example, you can use `[os.urandom](https://docs.python.org/2/library/os.html?highlight=os#os.urandom)` to generate a random string to be attached to the timestamp before hashing. The randomness of the value should be strong enough to guarantee the uniqueness. 

To be honest, I'm a bit surprised there are duplicate names given that the timestamp includes microseconds. What causes it? Are tests run in parallel?

**Comment by jaredmoo** [comment]
Agreed it's not ideal. I was also surprised, I'm guessing that timestamp doesn't have high enough resolution. I'll try urandom.

**Comment by troydai** [comment]
There are probably other reason for this name to be duplicated. The hash digest return string with characters in a range [0-9a-f], I think this can be replaced with base64 so that in a fixed length more information can be contained. I'll try to work out a prototype tonight.

**Comment by jaredmoo** [comment]
Ok. Can you try running my test in live mode with whatever approach you choose? 

The reason I think it's time resolution related is because the issue went away when I stepped through with debugger and it sometimes went away when I added some print statements. And this is probably the first time that a test case has two of the same Preparer and skips the slow API call in the Preparer's `create_resource` function.

**Comment by troydai** [comment]
Yes. I'll try it out.

File: src/command_modules/azure-cli-sql/tests/test_sql_commands.py [hunk 1]
```diff
@@ -58,8 +60,12 @@ def _get_resource_group(self, **kwargs):
 class SqlServerMgmtScenarioTest(ScenarioTest):
 
     @ResourceGroupPreparer()
-    def test_sql_server_mgmt(self, resource_group, resource_group_location):
-        servers = ['cliautomation53', 'cliautomation54']  # TODO: Server names should be randomized
+    # Skip create in SqlServerPreparer because we want to create in the test. So why even use
+    # SqlServerPreparer? We still want to generate random server name and replace it in the test
+    # recording, which SqlServerPreparer does very nicely.
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: For this cause, you don't need a preparer. The `create_random_name` method if `ScenarioTest` will register the random name with moniker in a name replacer. The replacer to alter the recording file.

Reference: https://github.com/Azure/azure-cli/blob/master/src/azure-cli-testsdk/azure/cli/testsdk/base.py#L173

Give it a try.

Comment (jaredmoo) [inline src/command_modules/azure-cli-sql/tests/test_sql_commands.py]: That worked. Much easier than I expected.

**Comment by troydai** [comment]
Looks great. Thanks, @jaredmoo.

**Pull request merged**